pipeline/backend/docker/docker.go

```diff
@@ -17,28 +17,31 @@ import (
 	backend "github.com/woodpecker-ci/woodpecker/pipeline/backend/types"
 )
 
-type engine struct {
+type docker struct {
 	client client.APIClient
 }
 
+// make sure docker implements Engine
+var _ backend.Engine = &docker{}
+
 // New returns a new Docker Engine.
 func New() backend.Engine {
-	return &engine{
+	return &docker{
 		client: nil,
 	}
 }
 
-func (e *engine) Name() string {
+func (e *docker) Name() string {
 	return "docker"
 }
 
-func (e *engine) IsAvivable() bool {
-	_, err := os.Stat("/.dockerenv")
-	return os.IsNotExist(err)
+func (e *docker) IsAvailable() bool {
+	_, err := os.Stat("/var/run/docker.sock")
```

Comment:
Strange I don't get why this wasn't working. I checked a clean container for that file. Using the docker sock for detection only works as we are mounting it to the agent, but should be fine for now.

Reply:
well `/.dockerenv` detect if agent is within an container ... but that was never what we should check for - we like to know it we have access to an docker daemon :)